circuit/src/rns/secp256k1.rs

```diff
@@ -0,0 +1,222 @@
+	// sanity check: https://www.wolframalpha.com/input?i=140029228562771870679+%2B+54120732105655028278+*+2%5E68+%2B+18037446069688272914+*+2%5E136+%2B+604307990016668+*+2%5E204
+	fn negative_wrong_modulus_decomposed() -> [Fr; 4] {
+		let limb0 = Fr::from_u128(140029228562771870679);
+		let limb1 = Fr::from_u128(54120732105655028278);
+		let limb2 = Fr::from_u128(18037446069688272914);
+		let limb3 = Fr::from_u128(604307990016668);
+		[limb0, limb1, limb2, limb3]
+	}
+
+	fn right_shifters() -> [Fr; 4] {
+		let limb0 = Fr::from_u128(1);
+		let limb1 = Fr::from_raw([
+			0xf8e4610fb396ee5, 0xb42e346981868e48, 0x1dbc9c192fc7933a, 0xb603a5609b3f6f8,
+		]);
+		let limb2 = Fr::from_raw([
+			0x568bea8e0766f9dd, 0xa31a140f219532a9, 0x1a908db2cea9b991, 0x1b7c016fe8acfaed,
+		]);
+		let limb3 = Fr::from_raw([
+			0x769b0bf04e2f27cc, 0x55a33201cd88df51, 0x338287b1e0bedd99, 0x523513296c10199,
+		]);
+		[limb0, limb1, limb2, limb3]
+	}
+
+	fn left_shifters() -> [Fr; 4] {
+		let limb0 = Fr::from_u128(1);
+		let limb1 = Fr::from_raw([0x0, 0x10, 0x0, 0x0]);
+		let limb2 = Fr::from_raw([0x0, 0x0, 0x100, 0x0]);
+		let limb3 = Fr::from_raw([0x0, 0x0, 0x0, 0x1000]);
+		[limb0, limb1, limb2, limb3]
+	}
+
+	fn to_add_x() -> [Fr; 4] {
+		let limb0 = Fr::from_u128(252992975607365614304);
+		let limb1 = Fr::from_u128(246373929609370463456);
+		let limb2 = Fr::from_u128(147045207087024714823);
+		let limb3 = Fr::from_u128(661255989794087);
+		[limb0, limb1, limb2, limb3]
+	}
+
+	fn to_add_y() -> [Fr; 4] {
+		let limb0 = Fr::from_u128(100703685036303013336);
+		let limb1 = Fr::from_u128(142420853763560729700);
+		let limb2 = Fr::from_u128(70276077464837018148);
+		let limb3 = Fr::from_u128(3853257862753614);
+		[limb0, limb1, limb2, limb3]
+	}
+
+	fn to_sub_x() -> [Fr; 4] {
+		let limb0 = Fr::from_u128(112383145895474475045);
+		let limb1 = Fr::from_u128(77353473636075571401);
+		let limb2 = Fr::from_u128(168420387443881604905);
+		let limb3 = Fr::from_u128(4333384540989656);
+		[limb0, limb1, limb2, limb3]
+	}
+
+	fn to_sub_y() -> [Fr; 4] {
+		let limb0 = Fr::from_u128(34847995348328062435);
+		let limb1 = Fr::from_u128(138213416034526311223);
+		let limb2 = Fr::from_u128(277306927096733657243);
+		let limb3 = Fr::from_u128(158537774519885);
+		[limb0, limb1, limb2, limb3]
+	}
+
+	fn invert(input: BigUint) -> Option<Integer<secp256k1Fp, Fr, 4, 68, Self>> {
+		let a_w = big_to_fe::<secp256k1Fp>(input);
+		let inv_w = a_w.invert();
+		inv_w
+			.map(|inv| {
+				Integer::<secp256k1Fp, Fr, 4, 68, Secp256k1BaseField4_68>::new(fe_to_big(inv))
+			})
+			.into()
+	}
+}
+
+#[derive(Debug, Clone, PartialEq, Default)]
+/// Struct for the Secp256k1 Scalar Field as the wrong field.
+/// From https://github.com/privacy-scaling-explorations/halo2curves/blob/main/src/secp256k1/fq.rs
+pub struct Secp256k1ScalarField4_68;
```

Comment:
```suggestion
pub struct Secp256k1_Scalar_4_68;
```

Reply:
The linter does not allow me to do this because of the `#[deny(non_camel_case_types)]` constraint.